charms/mlflow-server/src/charm.py

```diff
@@ -287,5 +268,45 @@ def __init__(self, msg, status_type=None):
         self.status = status_type(msg)
 
 
+def _b64_encode_dict(d):
+    """Returns the dict with values being base64 encoded."""
+    # Why do we encode and decode in utf-8 first?
+    return {k: b64encode(v.encode("utf-8")).decode("utf-8") for k, v in d.items()}
+
+
+def _minio_credentials_dict(obj_storage):
+    """Returns a dict of minio credentials with the values base64 encoded."""
+    minio_credentials = {
+        "AWS_ENDPOINT_URL": f"http://{obj_storage['service']}:{obj_storage['port']}",
+        "AWS_ACCESS_KEY_ID": obj_storage["access-key"],
+        "AWS_SECRET_ACCESS_KEY": obj_storage["secret-key"],
+        "USE_SSL": str(obj_storage["secure"]).lower(),
+    }
+    return _b64_encode_dict(minio_credentials)
+
+
+def _seldon_credentials_dict(obj_storage):
```

Comment:
how does do the seldon-init containers get these credentials?  Specially when the seldon deployment is in a different namespace.

Reply:
For now, only with some extra steps :(  There's more context in [this comment](https://github.com/canonical/bundle-kubeflow/issues/429#issuecomment-1055755776).  This is more a stop-gap fix until we have a way of injecting secrets into everyone's namespaces. 

This first step was decided on with @Barteus so that:
* for deployments with Seldon and MLFlow in the same namespace, `SeldonDeployment`s work easily by referencing this secret directly
* for deployments with Seldon and MLFlow in different namespaces, `SeldonDeployment`s **cannot** reference this secret directly but at least there's a "template" secret in mlflow's namespace that is easy to copy as needed

As soon as we can inject secrets into user namespaces, we remove this secret.

After talking with @Barteus, this secret means if seldon